src/Microsoft.Identity.Client/User.cs

```diff
@@ -47,37 +46,34 @@ internal User(User other)
             IdentityProvider = other.IdentityProvider;
         }
 
+        public User(string identifier, string displayableId, string name, string identityProvider)
+        {
+            if (string.IsNullOrWhiteSpace(identifier))
+            {
+                throw new ArgumentNullException(nameof(identifier));
+            }
+
+            DisplayableId = displayableId;
+            Name = name;
+            IdentityProvider = identityProvider;
```

Comment:
They don't need to create a dummy user object to be able to mock out a behavior of acquire token API. They can pass null for all they care into the mocked method. Adding interface to user object is just another level of indirection as it does not provide behavior.